backend/src/routes/v1/approvals.ts

```diff
@@ -132,6 +139,20 @@ export const postApprovalResponse = [
       version[field] = choice as ApprovalStates
       await version.save()
 
+      const { managerTitle, reviewerTitle } = await getManagerAndReviewer(version.metadata)
```

Comment:
There's no reason to call this every single time.  Only call it when an error is made by putting it inside the two `if` statements.